pvl/encoder.py

```diff
@@ -643,8 +678,14 @@ def encode_value(self, value):
         """Extends parent function by only allowing Units Expressions for
         numeric values.
         """
-        if(isinstance(value, Units) and
-           not isinstance(value.value, (int, float))):
+        if isinstance(value, tuple(x[0] for x in self.quantities)):
```

Comment:
there is something confusing about the logic here. It looks like we are iterating through self.quantities multiple times so this could probably be simplified/reworked

Reply:
Good catch, refactored.